src/basic/fd-util.c

```diff
@@ -805,6 +805,13 @@ int fd_reopen(int fd, int flags) {
                  * magic /proc/ directory, and make ourselves independent of that being mounted. */
                 return RET_NERRNO(openat(fd, ".", flags | O_DIRECTORY));
 
+        /* Keep/propagate O_APPEND, if set on original fd.
+         * Otherwise we get rather "unexpected" behavior,
+         * for example with repeated "systemd-run --pty >> some-log" */
+        int orig_status_flags = fcntl(fd, F_GETFL);
+        if (orig_status_flags != -1 && FLAGS_SET(orig_status_flags, O_APPEND))
```

Comment:
No. The whole point of `fd_reopen` is that we use new flags. We rely on the behavior internally.

For your use case, `--pipe` should be used, no?

Reply:
Sometimes you want to have a pty (propagation of tty signals), but still want to keep a log. The workaround now is to basically do `| tee -a some-log` within the actual command instead [or `chattr +a` ;-)]. And no, re-opening something that has O_APPEND set without it is "always nonsense", IMO.

compare the result of:
```sh
: > log
for i in 1 2 3 4 5; do
    date +%n%F_%T >> log;
    ( sleep 0.$RANDOM; seq 9 ) >> log &
    ( sleep 0.$RANDOM; seq 1001 1004 ) >> log &
    sleep 1
done
```
("expected" result in `log`)

and
```sh
: > log
for i in 1 2 3 4 5; do
    date +%n%F_%T >> log;
    ( sleep 0.$RANDOM; seq 9 | dd of=/proc/self/fd/1 conv=notrunc ) >> log &
    ( sleep 0.$RANDOM; seq 1001 1004  | dd of=/proc/self/fd/1 conv=notrunc ) >> log &
    sleep 1
done
```
(basically what systemd-run does now; "unexpected" result in log)